consensus/threshold.go

```diff
@@ -42,12 +44,16 @@ func (consensus *Consensus) didReachPrepareQuorum() error {
 	// so by this point, everyone has committed to the blockhash of this block
 	// in prepare and so this is the actual block.
 	for i, key := range consensus.PubKey.PublicKey {
-		consensus.Decider.SubmitVote(
+		if _, err := consensus.Decider.SubmitVote(
```

Comment:
We should be lenient on this. If the leader self vote is not successful, he should still try continue  the consensus with other validators.

Reply:
The only way it fails is on the Phase being unknown, which has never failed before